documentation/bin/createDocu.groovy

```diff
@@ -79,6 +79,97 @@ class TemplateHelper {
+'''
+        return t.trim()
+    }
+
+    static createStepActivationSection(Map configConditions) {
+        def t = 'Certain steps will be activated automatically depending on following conditions:\n\n'
+
+
+        t += '| step | config key | config value | file pattern |\n'
+        t += '| ---- | ---------- | ------------ | ------------ |\n'
+
+        configConditions?.each {stepName, conditions ->
+            t += "| ${stepName} "
+            t += "| ${renderValueList(conditions?.configKeys)} "
+            t += "| ${renderValueList(mapToValueList(conditions?.config))} "
+
+            List filePatterns = []
+            if (conditions?.filePattern) filePatterns.add(conditions?.filePattern)
+            if (conditions?.filePatternFromConfig) filePatterns.add(conditions?.filePatternFromConfig)
+            t += "| ${renderValueList(filePatterns)} |\n"
+        }
+
+        t += '''
+!!! info "Step condition details"
+    There are currently several conditions which can be checked.<br /> This is done in the [Init stage](init.md) of the pipeline shortly after checkout of the source code repository.<br/ >
+    **Important: It will be sufficient that any one condition per step is met.**
+    
+    * `config key`: Checks if a defined configuration parameter is set.
+    * `config value`: Checks if a configuration parameter has a defined value.
+    * `file pattern`: Checks if files according a defined pattern exist in the project. Either the pattern is speficified direcly or it is retrieved from a configuration parameter. 
+    
+
+!!! note "Overruling step activation conditions"
+    It is possible to overrule the automatically detected step activation status.<br />
+    Just add to your stage configuration `<stepName>: false`, for example `deployToKubernetes: false`.
+
+For details about the configuration options, please see [Configuration of Piper](../configuration.md).
+'''
+
+        return t.trim()
+    }
+
+    private static renderValueList(List valueList) {
+        if (!valueList) return ''
+        if (valueList.size() > 1) {
+            List quotedList = []
+            valueList.each {listItem ->
+                quotedList.add("-`${listItem}`")
+            }
+            return quotedList.join('<br />')
+        } else {
+            return "`${valueList[0]}`"
+        }
+    }
+
+    private static mapToValueList(Map map) {
+        List valueList = []
+        map?.each {key, value ->
+            if (value instanceof List) {
+                value.each {listItem ->
+                    valueList.add("${key}: ${listItem}")
+                }
+            } else {
+                valueList.add("${key}: ${value}")
+            }
+        }
+        return valueList
+    }
+
+    static createStageConfigurationSection() {
+        return 'The stage parameters need to be defined in the section `stages` of [config.yml file](../configuration.md).'
```

Comment:
I'm not aware of that `../config.yml`.

* Is this `.pipeline/config.yml`? 
* Is that a file expected to be present in the generated docu (more likely)? There is currently no such file.

Reply:
It's a reference to `../configuration.md` which is available ...